### PR TITLE
Northwest Territories Basic Personal Amount

### LINKED ISSUE #373: Northwest Territories Basic Personal Amount
Reference: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1nt/td1nt-23e.pdf (#1)

### PR DESCRIPTION
**Pull request by zqw5219**
Fixes #373

### CODE REVIEW

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+values:
+  2023-01-01: 16_593
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Northwest Territories Basic Personal Amount
+  reference:
+    - title: Northwest Territories Personal Tax Credits Return - Basic Personal amount
+      href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1nt/td1nt-23e.pdf
+    - title:
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml:11]: add title to reference

File: policyengine_canada/variables/gov/provinces/nt/tax/income/credits/basic_personal_amount/nt_basic_personal_amount.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+class nt_basic_personal_amount(Variable):
+    value_type = float
+    entity = Person
+    label = "Northwest Territories basic personal amount"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = ProvinceCode.NT
+    reference = "https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1nt/td1nt-23e.pdf"
+
+    adds = "gov.provinces.nt.tax.income.credits.basic_personal_amount.base"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nt/tax/income/credits/basic_personal_amount/nt_basic_personal_amount.py:13]: we have to make a formula where we check head * amount

Comment (PavelMakarchuk) [reply policyengine_canada/variables/gov/provinces/nt/tax/income/credits/basic_personal_amount/nt_basic_personal_amount.py:13]: then we need a unit test

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+description: Northwest Territories provides the following Basic Personal Amount.
+values:
+  2023-01-01: 16_593
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml:3]: need 2022 values and 2022 reference

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Northwest Territories Basic Personal Amount
+  reference:
+    - title: Northwest Territories Personal Tax Credits Return - Basic Personal Amount
+      href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1nt/td1nt-23e.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml:10]: need pdf pages in the hyperlink for all hrefs

File: policyengine_canada/tests/gov/provinces/nt/tax/income/nt_basic_personal_amout.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/nt/tax/income/nt_basic_personal_amout.yaml:1]: make one more test outside of NT

File: policyengine_canada/tests/gov/provinces/nt/tax/income/nt_basic_personal_amout.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    nt_basic_personal_amount: 0
+
+- name: 2023 NT basic personal amount
+  period: 2023
+  input:
+    province_code: NT
+  output:
+    nt_basic_personal_amount: 16_593
+
+- name: 2023 NT basic personal amount; wrong provience
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/nt/tax/income/nt_basic_personal_amout.yaml:22]:
```suggestion
- name: 2023 NT basic personal amount; wrong province
```

File: policyengine_canada/tests/gov/provinces/nt/tax/income/nt_basic_personal_amout.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+- name: 2022 NT basic personal amount
+  period: 2022
+  input:
+    province_code: NT
+  output:
+    nt_basic_personal_amount: 15_609
+
+- name: 2022 NT basic personal amount; wrong provience
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/nt/tax/income/nt_basic_personal_amout.yaml:8]:
```suggestion
- name: 2022 NT basic personal amount; wrong province
```

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+values:
+  2022-01-01: 15_609
+  2023-01-01: 16_593
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Northwest Territories Basic Personal Amount
+  reference:
+    - title: 2022 Northwest Territories Personal Tax Credits Return - Basic Personal Amount
+      href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5012-c/5012-c-22e.pdf #page=1
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml:11]:
```suggestion
      href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5012-c/5012-c-22e.pdf#page=1
```

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    - title: 2023 Northwest Territories Personal Tax Credits Return - Basic Personal Amount
+      href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1nt/td1nt-23e.pdf #page=1
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml:13]:
```suggestion
      href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1nt/td1nt-23e.pdf#page=1
```

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml [hunk 9]
```diff
@@ -0,0 +1,13 @@
+description: Northwest Territories provides the following Basic Personal Amount.
+values:
+  2022-01-01: 15_609
+  2023-01-01: 16_593
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Northwest Territories Basic Personal Amount
+  reference:
+    - title: 2022 Northwest Territories Personal Tax Credits Return - Basic Personal Amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/basic_personal_amount/base.yaml:10]: need the income tax act reference + section

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_canada/tests/gov/provinces/nt/tax/income/nt_basic_personal_amout.yaml
Comment (MaxGhenis) [inline policyengine_canada/tests/gov/provinces/nt/tax/income/nt_basic_personal_amout.yaml:1]: rename amount

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]
Per Pavel and my discussion, merging this for now but would suggest hooking these variables up to the net income tree moving forward

**Pull request merged**